docs/developer/extending/api/sorters.mdx

```diff
@@ -0,0 +1,117 @@
+    NAME = ["name", "slug"]
+    VISIBILITY = ["is_published", "name", "slug"]
+
+    @property
+    def description(self):
+        if self.name in ExampleSortField.__enum__._member_names_:
+            sort_name = self.name.lower().replace("_", " ")
+            return f"Sort Example by {sort_name}."
+        raise ValueError("Unsupported enum value: %s" % self.value)
+```
+
+GraphQL field name defines the schema field, which can be used to sort elements,
+and value defines the list of model fields, that will be used for sorting. The list can contain
+multiple elements, the next value from the list will be used when items have the same value
+for previous field from the list. For example, if elements sorting by `NAME` have the same
+value of the `name` field, they will be sorted by `slug` field value.
+
+### Step 2: Create Sort Input
+In the same file create Sorting Input with `Meta`. In `Meta` define `sort_enum` which
+corresponds to just created sorted enum and `type_name` which is used to create sorter description.
+
+```python
+from saleor.graphql.core.types import SortInputObjectType
+
+
+class ExampleSortingInput(SortInputObjectType):
+    class Meta:
+        sort_enum = ExampleSortField
+        type_name = "examples"
+```
+
+### Step 3: Add sorter to query
+Define the `sort_by` field in your target query resolver in the `schema.py` file. Make sure that resolver use
+`BaseDjangoConnectionField` or another type which inherits from this. Here is an example:
+
+```python {9}
+import graphene
+from saleor.graphql.core.field import BaseDjangoConnectionField
+
+
+class ExampleQueries(graphene.ObjectType):
+    examples = BaseDjangoConnectionField(
+        Example,
+        description="List of the examples.",
+        sort_by=ExampleSortingInput(description="Sorting options for examples.")
+    )
+```
+
+## Add custom sorter
+When defining sorter apart from a model field you can define the custom one, and use it
+like other fields.
+
+### Step 1: Define static method in your sorter field
+The method name must be in format `qs_with_new_field` where `new_field` corresponds
+to the custom field which you wanted to sort your queryset. The method must take the
+queryset as an argument and return queryset with the annotated new field.
+
+
+```python {12-14}
+class ExampleSortField(graphene.Enum):
+    NAME = ["name", "slug"]
+    VISIBILITY = ["is_published", "name", "slug"]
+
+    @property
+    def description(self):
+        if self.name in ExampleSortField.__enum__._member_names_:
+            sort_name = self.name.lower().replace("_", " ")
+            return f"Sort Example by {sort_name}."
+        raise ValueError("Unsupported enum value: %s" % self.value)
+
+    @staticmethod
+    def qs_with_item_count(queryset: Queryset) -> Queryset:
+        return queryset.annotate(item_count=Count("item"))
+```
+
+### Step 2: Use the field in sort field definition
+The next step is just using it in sort field definition as a normal model field.
```

Comment:
Maby we should add some description of that value from (`ITEM_COUNT = ["item_count"]`) this list should be the same as an annotated field name?

Reply:
I added notes in this and the previous section.